# Import packages

In [18]:
from superQuery import SuperQuery
from itertools import islice
import json

# Create superQuery instance

In [19]:
sq = SuperQuery()

# Get your data

In [28]:
# We don't want to do a dry-run
dryrun = False

mydata = sq.get_data(
    """SELECT field1, field2 FROM `projectId.datasetId.tableId` WHERE _PARTITIONTIME = \"2019-05-04\" ORDER BY field1 ASC""", 
    dry_run = dryrun,
    username="xxxxxx", 
    password="xxxxxx") # Get your login details by signing into superquery.io and clicking on Integrations.

[sQ]...Connection to superQuery successful


# Let's explore your data

## Stats

In [31]:
if (not dryrun):
    print("Data rows:", mydata.totalRows)
    print("Workflow:", "Query")
    print("Cost:", mydata.superQueryTotalCost)
    print("Savings %:", mydata.saving)
    print("Was cache used?:", mydata.cacheUsed if hasattr(mydata, "cacheUsed") else False)
    print("DryRun flag: ", mydata.superParams["isDryRun"])
else:
    print("Execution type:", "DryRun")
    print("Potential BQ bytes scanned: ", mydata.bigQueryTotalBytesProcessed)
    print("Potential Data rows:", mydata.totalRows)
    print("Cost:", mydata.superQueryTotalCost)
    print("DryRun flag: ", mydata.superParams["isDryRun"])

Execution type: DryRun
Potential BQ bytes scanned:  273619
Potential Data rows: 8963
Cost: 0
DryRun flag:  False


## Let's get only a specific amount of rows

### First 6k rows

In [26]:
first6k_rows = list(islice(mydata, 6000))  
print(len(first6k_rows))

6000


## Can we get another 6k rows? No, we only have 8963 in total

In [27]:
next6k_rows = list(islice(mydata, 6000))  
print(len(next6k_rows))

2963
